microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/spi/sender/WildFlySenderFactory.java

```diff
@@ -30,7 +32,12 @@ public class WildFlySenderFactory implements SenderFactory {
 
     @Override
     public Sender getSender(Configuration.SenderConfiguration configuration) {
-        return new WildFlySender(delegate.getSender(configuration));
+        if (configuration.getEndpoint() == null) {
```

Comment:
You need to check configuration.getAgentHost() and configuration.getAgentPort() as well. ThriftSenderFactory uses those to create a UdpSender is there is no endpoint is configured. Those are configured via the sender-binding attribute in the subsystem model.

Tangent: SENDER_BINDING should not allow expressions. It's a capability reference and those shouldn't allow expressions.